relay-metrics/src/aggregator.rs

```diff
@@ -208,7 +202,7 @@ pub enum ShiftKey {
     Bucket,
 }
 
-/// Parameters used by the [`AggregatorService`].
+/// Parameters used by the [`Aggregator`].
 #[derive(Clone, Debug, Deserialize, Serialize)]
 #[serde(default)]
 pub struct AggregatorConfig {
```

Comment:
We could split this into `AggregatorConfig` and `AggregatorServiceConfig`, and then `#[serde(flatten)]` them into a single object for the serialized config.

Reply:
id have to make a third config to split and merge from for that right?

Reply:
Something like this should work:

```rust
struct AggregatorServiceConfig {
    #[serde(flatten)]
    aggregator: AggregatorConfig,
    some_service_related_field: u64,
}

Reply:
would this serviceconfig actually be used directly or would it pass the service related fields to the service and the aaggregator config to the aggregator?

Reply:
The `AggregatorService` would get the `AggregatorServiceConfig`, and delegate the `AggregatorConfig` to the `Aggregator`. That way, every component gets only the config that is relevant to that component and nothing else.